os/board/rtl8721csm/src/component/os/tizenrt/tizenrt_service.c

```diff
@@ -851,13 +894,54 @@ _timerHandle _tizenrt_timerCreate(const signed char *pcTimerName, osdepTickType
 	timer->timevalue = xTimerPeriodInTicks;
 	timer->data = pvTimerID;
 	timer->function = pxCallbackFunction;
+
+	if (!_tizenrt_timer_table_init) {
+		INIT_LIST_HEAD(&_tizenrt_timer_table);
+		_tizenrt_timer_table_init = 1;
+	}
+
+	struct _tizenrt_timer_entry *timer_entry;
+	timer_entry = _tizenrt_zmalloc(sizeof(struct _tizenrt_timer_entry));
+	if (timer_entry == NULL) {
+		kmm_free(timer->work_hdl);
+		kmm_free(timer);
+		return NULL;
+	}
+	memset(timer_entry, 0, sizeof(struct _tizenrt_timer_entry));
+	timer_entry->timer = timer;
+
+	save_and_cli();
```

Comment:
To manage timer list, do you really need interrupt lock?? not like sem or mutex?

Reply:
We have been using it all along, we dont have intention to change it currently. Any issue?

Reply:
As you know, irq lock will cause irq missing. Because of irq missing, Realtek optimized irq usages like flash operation. I think this is not more important than flash operation. And there is no reason to use irqlock instead of sem. Could you let me know the reason why this is necessary?

Reply:
I have change irqlock to sem/mutex to avoid irq missing.
By using sem/mutex it will have some drop for uplink t-put about 1Mbits/sec drops.
Because sem/mutex take more processing time, in lib_wlan all timer are using it.
Please take note about his.

Reply:
@pillip8282 Could you check @ziyik 's comment? Originally it uses irq lock / unlock to manage common resource. But it will causes irq missing so that I asked to use sem or mutex. But he said it will cause wifi throughput performance drop. Please let us know your opinion.